udata/api/oauth2.py

```diff
@@ -324,13 +335,13 @@ def query_client(client_id):
 
 def save_token(token, request):
     scope = token.pop("scope", "")
+    client = request.client
+    user = request.user or client.owner
     if request.grant_type == "refresh_token":
-        credential = request.credential
-        credential.update(scope=scope, **token)
+        old_token = OAuth2Token.objects(client=client, user=user, scope=scope).first()
```

Comment:
Shouldn't `refresh_token` param be the identifier? Ex: `OAuth2Token.objects(refresh_token=request.refresh_token).first()` or similar?

Reply:
Very good point indeed! I modified the code, and the tests to make sure we only update this token, and not any other token.

I've even tested that if a user has two tokens (which shouldn't happen?), only the correct one is refreshed.